SEED-platform/seed
# Improve handling of failed ESPM imports

#### Any background context you want to provide?
If an org MapQuest API key was invalid, the import process would stall and the mapping suggestions would not be saved.

#### What's this PR do?
- puts a catch statement on the `uploader_factory.check_progress_loop_main_sub` function.
- always display existing mapping suggestions, even if matching had not previously completed
- translations
- refactors for clarity

#### How should this be manually tested?
Upload with an invalid MapQuest API.
1. Enter an invalid MapQuest API key. (if the API key is blank, geocoding is skipped)
2. Upload ESPM data: Data > Import Portfolio Manager Data > Log in with SEED_City_test > select SEED City Benchmarking Report > go through mapping/matching.
3. User should be presented with an informative error message.
4. After the failed import, go to Data > click on the dataset name > Data Paring for the failed import > the mappings should appear but are still editable as it was unsuccessful.

#### What are the relevant tickets?
#4675 #4674


#### Screenshots (if appropriate)
<img width="599" alt="Screenshot 2024-08-27 at 1 41 53 PM" src="https://github.com/user-attachments/assets/2d1316df-3449-43de-bfb4-7724209d5546">
